plugin/evm/limitorders/liquidations.go

```diff
@@ -0,0 +1,62 @@
+package limitorders
+
+import (
+	"math"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const maintenanceMargin float64 = 0.1
+
+type Liquidable struct {
+	Address common.Address
+	Size    float64
+}
+
+func (db *InMemoryDatabase) GetLiquidableTraders(market Market, markPrice float64, oraclePrice float64) []Liquidable {
+
+	toLiquidate := []Liquidable{}
+
+	overSpreadLimit := isOverSpreadLimit(markPrice, oraclePrice)
+
+	for addr, trader := range db.GetAllTraders() {
+		position := trader.Positions[market]
+		notionalPosition := position.Size * markPrice
+		var unrealisedPnL float64
+		if position.Size > 0 {
+			unrealisedPnL = notionalPosition - position.OpenNotional
+		} else {
+			unrealisedPnL = position.OpenNotional - notionalPosition
+		}
+
+		margin := trader.GetNormalisedMargin() + position.UnrealisedFunding
+		marginFraction := (margin + unrealisedPnL) / notionalPosition
+		if overSpreadLimit {
+			var oracleBasedUnrealizedPnl float64
+
+			oracleBasedNotional := oraclePrice * math.Abs(position.Size)
+			if position.Size > 0 {
+				oracleBasedUnrealizedPnl = oracleBasedNotional - position.OpenNotional
+			} else if position.Size < 0 {
+				oracleBasedUnrealizedPnl = position.OpenNotional - oracleBasedNotional
+			}
+
+			marginFraction = (margin + oracleBasedUnrealizedPnl) / oracleBasedNotional
```

Comment:
I think when the amm isOverSpreadLimit we use the max of 2 marginFractions. @asquare08 can confirm and infact even take a shot at writing this code.

Reply:
Can you explain logic behind this whole thing ?
Why we max only in case of isOverSpreadLimit ?

Reply:
When there is a significant difference between markPrice and oraclePrice, we don't want our users to get liquidated if any of the prices saves them. We liquidate only when they are in the liquidation zone by both prices. This is a safeguard against liquidations by price manipulations. And when mark and oracle prices are close to each other, mark price is used as usual.